src/main/java/xbot/common/trajectory/LowResField.java

```diff
@@ -24,13 +24,306 @@ public LowResField() {
+        // New attempt at a pathfinding algorithm that maybe a little more resilient. The original algorithm had
+        // two issues:
+        // - It worked backwards, which was a little unintuitive, but not strictly a problem.
+        // - It didn't reliably check for new collisions when generating intermediate points.
+        // This approach should solve both those issues. Basic idea:
+        // - Check for collisions between robot and target.
+        // - If collision, set the nearest corner of that obstacle as new target.
+        // - Check to make sure that there are no obstacles between you and that corner - if so, repeat the above.
+        // - Once satisfied there are no obstacles, advance to the corner and repeat the process of trying to get
+        //   to the ultimate target.
+
+        // General code structure::
+        // 1) Save the initial target as the ultimate target; we'll need it later.
+        //      a) initialize the current target from the ultimate target.
+        //      b) initialize an empty list of swerve points.
+        // 2) Create a Stack of XbotSwervePoints, which will be used to help us deal with multiple intermediate points.
+        // ===== Main loop =====
+        // 3) Raycast to the center of each obstacle, recording distance to the first intersection with the bounding box.
+        //      a) Put the obstacles into a TreeMap sorted by that distance.
+        // 4) Raycast from the current position to the destination, and iterate through each of the obstacles to see
+        //    if there is a collision.
+        //      a) If no collision, AND the current target is the ultimate target, we are done.
+        //         Add the current target to the list, and return.
+        //      b) If no collision, and the stack isn't empty, fully pop the stack into the list and set
+        //         the current target to the ultimate target, and the current position to the last point in the list.
+        //         Loop back to step (3).
+        //      c) If collision, get the nearest legal corner, put it on the stack, and set it as the current target.
+        //         Loop back to step (3).
+
+        // 1) Save the initial target as the ultimate target; we'll need it later.
+        XbotSwervePoint ultimateTarget = targetPoint;
+        //      a) initialize the current target from the ultimate target.
+        XbotSwervePoint currentTarget = ultimateTarget;
+        Translation2d currentSource = currentPose.getTranslation();
+        //      b) initialize an empty list of swerve points.
+        List<XbotSwervePoint> path = new ArrayList<XbotSwervePoint>();
+        // 2) Create a Stack of XbotSwervePoints, which will be used to help us deal with multiple intermediate points.
+        Stack<XbotSwervePoint> swervePointStack = new Stack<XbotSwervePoint>();
+
+        XbotSwervePoint specialFinalPoint = null;
+
+        for (Obstacle o : obstacles) {
+            o.resetCorners();
```

Comment:
it feels a little icky to me for this function to have side effects on obstacles, maybe this should be working on a copy of them?